addons/l10n_pk_edi/models/account_move.py

```diff
@@ -0,0 +1,568 @@
+                error_msg
+            ))
+        for move in self:
+            if (move.move_type in ['in_refund', 'out_refund']
+                and (not move.l10n_pk_edi_cancel_reason or not move.l10n_pk_edi_cancel_remarks)
+            ):
+                corrupt_move_name[move.name] = 'error'
+        if corrupt_move_name:
+            error_msg = "\n".join(f"- {name}" for name, msg in corrupt_move_name.items())
+            raise ValidationError(_(
+                "To confirm credit note, set cancel reason and remarks at Electronic Invoicing tab in following invoices:\n%s",
+                error_msg
+            ))
+        return super()._post(soft)
+
+    def action_l10n_pk_edi_send(self):
+        """
+            Create a attachment and Send the invoice to the FBR.
+        """
+        self.ensure_one()
+        attachment_vals = self._l10n_pk_edi_get_attachment_values()
+        self.env['ir.attachment'].create(attachment_vals)
+        self.invalidate_recordset(fnames=['l10n_pk_edi_attachment_id', 'l10n_pk_edi_attachment_file'])
+        self.message_post(attachment_ids=self.l10n_pk_edi_attachment_id.ids)
+        self._l10n_pk_edi_send({self: attachment_vals})
+        self.is_move_sent = True
+
+    # -------------------------------------------------------------------------
+    # E-Invocing Methods
+    # -------------------------------------------------------------------------
+
+    def _get_l10n_pk_sales_groups(self):
+        return self.env['account.tax.group'].browse([
+            sale_group_id
+            for ref in (
+                'tax_group_pk_0',
+                'tax_group_pk_1',
+                'tax_group_pk_2',
+                'tax_group_pk_3',
+                'tax_group_pk_5',
+                'tax_group_pk_8',
+                'tax_group_pk_10',
+                'tax_group_pk_13',
+                'tax_group_pk_15',
+                'tax_group_pk_16',
+                'tax_group_pk_17',
+                'tax_group_pk_195',
+            )
+            if (sale_group_id := self.env['ir.model.data']._xmlid_to_res_model_res_id(
+                f"account.{self.company_id.id}_{ref}"
+            )[1])
+        ])
+
+    def _get_l10n_pk_withhold_groups(self):
+        return self.env['account.chart.template'].with_company(self.company_id).ref('tax_group_pk_wt')
+
+    def _l10n_pk_edi_get_default_enable(self):
+        self.ensure_one()
+        is_taxable = (
+            self._get_l10n_pk_sales_groups()
+            & self.line_ids.tax_ids.tax_group_id
+        )
+        return (
+            self.state == 'posted'
+            and self.company_id.account_fiscal_country_id.code == 'PK'
+            and self.journal_id.type in ['sale', 'purchase']
+            and self.l10n_pk_edi_state in (False, 'rejected')
+            and is_taxable
+        )
+
+    def _l10n_pk_validate_move_for_fbr(self):
+        error_messages = []
+        for line in self.invoice_line_ids.filtered(
```

Comment:
```py
        error_messages = []
        error_lines_name = {
            'negative_discount': [],
            'sale_type': [],
            'schedule_type': [],
            'hs_code_missing': [],
            'hs_code_invalid': []
        }
        for line in self.invoice_line_ids.filtered(
            lambda line: line.display_type == 'product'
            and not self._l10n_pk_is_global_discount(line)
        ):
            if line.discount < 0:
                error_lines_name['negative_discount'].append(line.name)
            if not line.product_id.l10n_pk_edi_sale_type_id:
                error_lines_name['sale_type'].append(line.name)
            if not line.product_id.l10n_pk_edi_schedule_code_id:
                error_lines_name['schedule_type'].append(line.name)
            hs_code = self._l10n_pk_edi_extract_digits(line.product_id.hs_code)
            if not hs_code:
                error_lines_name['hs_code_missing'].append(line.name)
            elif not re.match("^[a-zA-Z0-9]{8}$", hs_code):
                error_lines_name['hs_code_invalid'].append(line.product_id.name or line.name)
        if lines_name := error_lines_name['negative_discount']:
            error_messages.append(_("- Negative discount is not allowed, set in lines \n%s\n", '\n'.join(lines_name)))
        if lines_name := error_lines_name['sale_type']:
            error_messages.append(_("- Sale type is not defined for line \n%s\n", '\n'.join(lines_name)))
        if lines_name := error_lines_name['schedule_type']:
            error_messages.append(_("- Schedule code is not defined for line \n%s\n", '\n'.join(lines_name)))
        if lines_name := error_lines_name['hs_code_missing']:
            error_messages.append(_("- HS code is not set in product line \n%s\n", '\n'.join(lines_name)))
        if lines_name := error_lines_name['hs_code_invalid']:
            error_messages.append(_(
                "- Invalid HS Code in product line %(product_line)s it must be of 8 digits\n",
                    product_line='\n'.join(lines_name)
                )
            )
        return error_messages
```
Before:
![image](https://github.com/user-attachments/assets/b87fa44a-8d13-4c12-b0c3-c5a4f3ffca1d)
After:
![image](https://github.com/user-attachments/assets/42e1d9d8-04b8-448a-bac2-7894e04984cb)

@masm-odoo @lordkrandel @jco-odoo I believe the current error messages aren't that ideal for a user readability, well I have one suggestion. Let me know your opinion for this, feel free to optimize it if you have better suggestion (small addition to it we can try to make the error titles bold by using Markup 🤔 )

Reply:
In the Print and Send dialog we use something like this 
![immagine](https://github.com/user-attachments/assets/2986970f-bf9e-49c4-96dc-7af57ccabea0)
See: https://github.com/odoo/odoo/pull/142596

Reply:
There is a function in the wizard that returns a dict {'error_key': <warning_information>}, see this override here https://github.com/odoo/odoo/pull/166733/files#diff-a93d4442f33dead3e87d7e0e4ef601e1aa14bd48d6810555cf3354a5003c3fe1R46